File: APPManage/AM.Model/building_hoist.cs
/**  版本信息模板在安装目录下，可自行修改。
* building_hoist.cs
*
* 功 能： N/A
* 类 名： building_hoist
*
* Ver    变更日期             负责人  变更内容
* ───────────────────────────────────
* V0.01  2016/8/12 22:32:17   N/A    初版
*
* Copyright (c) 2012 Maticsoft Corporation. All rights reserved.
*┌──────────────────────────────────┐
*│　此技术信息为本公司机密信息，未经本公司书面同意禁止向第三方披露．　│
*│　版权所有：动软卓越（北京）科技有限公司　　　　　　　　　　　　　　│
*└──────────────────────────────────┘
*/
using System;
namespace AM.Model
{
	/// <summary>
	/// building_hoist:实体类(属性说明自动提取数据库字段的描述信息)
	/// </summary>
	[Serializable]
	public partial class building_hoist
	{
		public building_hoist()
		{}
		#region Model
		private string _guid;
		private string _userid;
		private DateTime? _createdate;
		private string _gcmc;
		private string _sydw;
		private string _sgdd;
		private string _jldw;
		private string _azdw;
		private string _azfzr;
		private string _sbmc;
		private string _sbxh;
		private string _zzdw;
		private string _zzxkzh;
		private string _ccbh;
		private string _azgzrq;
		private string _babh;
		private string _edzzl;
		private string _dlzl;
		private string _edtssd;
		private string _fzaqqxs;
		private string _dzzl;
		private string _tysazgd;
		private string _zydgd;
		private string _fqds;
		private string _dyfqgd;
		private string _bzjgg;
		private string _jylx;
		private string _yqmc1;
		private string _xh1;
		private string _bh1;
		private string _zt1;
		private string _yqmc2;
		private string _xh2;
		private string _bh2;
		private string _zt2;
		private string _yqmc3;
		private string _xh3;
		private string _bh3;
		private string _zt3;
		private string _yqmc4;
		private string _xh4;
		private string _bh4;
		private string _zt4;
		private string _yqmc5;
		private string _xh5;
		private string _bh5;
		private string _zt5;
		private string _bzxmbhgs;
		private string _ybxmbhgs;
		/// <summary>
		/// 
		/// </summary>
		public string Guid
		{
			set{ _guid=value;}
			get{return _guid;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string userid
		{
			set{ _userid=value;}
			get{return _userid;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime? CreateDate
		{
			set{ _createdate=value;}
			get{return _createdate;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string GCMC
		{
			set{ _gcmc=value;}
			get{return _gcmc;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SYDW
		{
			set{ _sydw=value;}
			get{return _sydw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SGDD
		{
			set{ _sgdd=value;}
			get{return _sgdd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string JLDW
		{
			set{ _jldw=value;}
			get{return _jldw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZDW
		{
			set{ _azdw=value;}
			get{return _azdw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZFZR
		{
			set{ _azfzr=value;}
			get{return _azfzr;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SBMC
		{
			set{ _sbmc=value;}
			get{return _sbmc;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SBXH
		{
			set{ _sbxh=value;}
			get{return _sbxh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZZDW
		{
			set{ _zzdw=value;}
			get{return _zzdw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZZXKZH
		{
			set{ _zzxkzh=value;}
			get{return _zzxkzh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string CCBH
		{
			set{ _ccbh=value;}
			get{return _ccbh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZGZRQ
		{
			set{ _azgzrq=value;}
			get{return _azgzrq;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BABH
		{
			set{ _babh=value;}
			get{return _babh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string EDZZL
		{
			set{ _edzzl=value;}
			get{return _edzzl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string DLZL
		{
			set{ _dlzl=value;}
			get{return _dlzl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string EDTSSD
		{
			set{ _edtssd=value;}
			get{return _edtssd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string FZAQQXS
		{
			set{ _fzaqqxs=value;}
			get{return _fzaqqxs;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string DZZL
		{
			set{ _dzzl=value;}
			get{return _dzzl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string TYSAZGD
		{
			set{ _tysazgd=value;}
			get{return _tysazgd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYDGD
		{
			set{ _zydgd=value;}
			get{return _zydgd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string FQDS
		{
			set{ _fqds=value;}
			get{return _fqds;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string DYFQGD
		{
			set{ _dyfqgd=value;}
			get{return _dyfqgd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BZJGG
		{
			set{ _bzjgg=value;}
			get{return _bzjgg;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string JYLX
		{
			set{ _jylx=value;}
			get{return _jylx;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YQMC1
		{
			set{ _yqmc1=value;}
			get{return _yqmc1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string XH1
		{
			set{ _xh1=value;}
			get{return _xh1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BH1
		{
			set{ _bh1=value;}
			get{return _bh1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZT1
		{
			set{ _zt1=value;}
			get{return _zt1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YQMC2
		{
			set{ _yqmc2=value;}
			get{return _yqmc2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string XH2
		{
			set{ _xh2=value;}
			get{return _xh2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BH2
		{
			set{ _bh2=value;}
			get{return _bh2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZT2
		{
			set{ _zt2=value;}
			get{return _zt2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YQMC3
		{
			set{ _yqmc3=value;}
			get{return _yqmc3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string XH3
		{
			set{ _xh3=value;}
			get{return _xh3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BH3
		{
			set{ _bh3=value;}
			get{return _bh3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZT3
		{
			set{ _zt3=value;}
			get{return _zt3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YQMC4
		{
			set{ _yqmc4=value;}
			get{return _yqmc4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string XH4
		{
			set{ _xh4=value;}
			get{return _xh4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BH4
		{
			set{ _bh4=value;}
			get{return _bh4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZT4
		{
			set{ _zt4=value;}
			get{return _zt4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YQMC5
		{
			set{ _yqmc5=value;}
			get{return _yqmc5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string XH5
		{
			set{ _xh5=value;}
			get{return _xh5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BH5
		{
			set{ _bh5=value;}
			get{return _bh5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZT5
		{
			set{ _zt5=value;}
			get{return _zt5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BZXMBHGS
		{
			set{ _bzxmbhgs=value;}
			get{return _bzxmbhgs;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YBXMBHGS
		{
			set{ _ybxmbhgs=value;}
			get{return _ybxmbhgs;}
		}
		#endregion Model

	}
}

File: AM/Scripts/appm/building_hoist.js
function testbh() {
    var isok = false;
    if ($("#gcname").val() == "") {
        alert("工程名称不能为空");
        $("#gcname").focus();
    } else {
        if ($("#sycompany").val() == "") {
            alert("使用单位不能为空");
            $("#sycompany").focus();
        } else {
            if ($("#sgaddress").val() == "") {
                alert("施工地点不能为空");
                $("#sgaddress").focus();
            } else {
                if ($("#jlcompany").val() == "") {
                    alert("监理单位不能为空");
                    $("#jlcompany").focus();
                } else {
                    if ($("#azcompany").val() == "") {
                        alert("安装单位不能为空");
                        $("#azcompany").focus();
                    } else {
                        if ($("#azpeople").val() == "") {
                            alert("安装负责人不能为空");
                            $("#azpeople").focus();
                        } else {
                            if ($("#sbname").val() == "") {
                                alert("设备名称不能为空");
                                $("#sbname").focus();
                            } else {
                                if ($("#type").val() == "") {
                                    alert("设备型号不能为空");
                                    $("#type").focus();
                                } else {
                                    if ($("#zzcompany").val() == "") {
                                        alert("制造单位不能为空");
                                        $("#zzcompany").focus();
                                    } else {
                                        if ($("#azdate").val() == "") {
                                            alert("安装告知日期不能为空");
                                            $("#azdate").focus();
                                        } else {
                                            if ($("#cccode").val() == "") {
                                                alert("出厂编号不能为空");
                                                $("#cccode").focus();
                                            } else {
                                                if ($("#xkzcode").val() == "") {
                                                    alert("制造许可证编号不能为空");
                                                    $("#xkzcode").focus();
                                                } else {
                                                    if ($("#ccdate").val() == "") {
                                                        alert("出厂日期不能为空");
                                                        $("#ccdate").focus();
                                                    } else {
                                                        if ($("#bacode").val() == "") {
                                                            alert("备案编号不能为空");
                                                            $("#bacode").focus();
                                                        } else {
                                                            if ($("#maxweight").val() == "") {
                                                                alert("最大额定起重量不能为空");
                                                                $("#maxweight").focus();
                                                            } else {
                                                                if ($("#edlj").val() == "") {
                                                                    alert("额定力矩不能为空");
                                                                    $("#edlj").focus();
                                                                } else {
                                                                    if ($("#maxspeed").val() == "") {
                                                                        alert("小车最大变幅速度不能为空");
                                                                        $("#maxspeed").focus();
                                                                    } else {
                                                                        if ($("#stnumber").val() == "") {
                                                                            alert("标准节规格/数量不能为空");
                                                                            $("#stnumber").focus();
                                                                        } else {
                                                                            if ($("#checkheight").val() == "") {
                                                                                alert("检验时安装高度不能为空");
                                                                                $("#checkheight").focus();
                                                                            } else {
                                                                                if ($("#maxrange").val() == "") {
                                                                                    alert("最大幅度/安装幅度不能为空");
                                                                                    $("#maxrange").focus();
                                                                                } else {
                                                                                    if ($("#yxheight").val() == "") {
                                                                                        alert("允许独立高度/允许自由端高度不能为空");
                                                                                        $("#yxheight").focus();
                                                                                    } else {
                                                                                        if ($("#checkspace").val() == "") {
                                                                                            alert("检验时安装附着道数/间距不能为空");
                                                                                            $("#checkspace").focus();
                                                                                        } else {
                                                                                            if ($("#wallheight").val() == "") {
                                                                                                alert("第一道附墙高度不能为空");
                                                                                                $("#wallheight").focus();
                                                                                            } else {
                                                                                                if ($("#checktype").val() == "") {
                                                                                                    alert("检验类别不能为空");
                                                                                                    $("#checktype").focus();
                                                                                                } else
                                                                                                    isok = true;
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
    return isok;
}
String.prototype.trim = function () {

    return this.replace(/(^\s*)|(\s*$)/g, '');
}

function submit() {
    var check = testbh();
    if (check)
    {
    var userid = $("#userid").val();
    var objArr = $("input[type=text]");
    var mobjArr1 = $("input[mtype=1]");
    //var checkitem = $("input[name=checkbox_name]:checked");
    //var radios = $("input[name=sex]:checked");
    //var xb = $(radios[0]).next("label").text();
    //var textareas = $("textarea");
    //var pxd = "";
    //for (var i = 0; i < checkitem.length; i++) {
        
    //    pxd += $(checkitem[i]).parent("label").text().trim();
    //    if (i < checkitem.length) {
    //        pxd += ","
    //    }
    //}
    var json = {
        "building_hoist": [{
            "Guid": "",
            "CreateDate": "",
            "userid": userid,
            "GCMC": $(objArr[0]).val(),
            "SYDW": $(objArr[1]).val(),
            "SGDD": $(objArr[2]).val(),
            "JLDW": $(objArr[3]).val(),
            "AZDW": $(objArr[4]).val(),
            "AZFZR" : $(objArr[5]).val(),
            "SBMC": $(objArr[6]).val(),
            "SBXH": $(objArr[7]).val(),
            "ZZDW": $(objArr[8]).val(),
            "ZZXKZH": $(objArr[9]).val(),
            "CCBH": $(objArr[10]).val(),
            "AZGZRQ": $(objArr[11]).val(),
            "BABH": $(objArr[12]).val(),
            "EDZZL": $(objArr[13]).val(),
            "DLZL": $(objArr[14]).val(),
            "EDTSSD": $(objArr[15]).val(),
            "FZAQQXS": $(objArr[16]).val(),
            "DZZL": $(objArr[17]).val(),
            "TYSAZGD": $(objArr[18]).val(),
            "ZYDGD": $(objArr[19]).val(),
            "FQDS": $(objArr[20]).val(),
            "DYFQGD": $(objArr[21]).val(),
            "BZJGG": $(objArr[22]).val(),
            "JYLX": $(objArr[23]).val(),
            "YQMC1": $(mobjArr1[0]).val(),
            "XH1": $(mobjArr1[1]).val(),
            "BH1": $(mobjArr1[2]).val(),
            "ZT1": $(mobjArr1[3]).val(),
            "YQMC2": $(mobjArr1[4]).val(),
            "XH2": $(mobjArr1[5]).val(),
            "BH2": $(mobjArr1[6]).val(),
            "ZT2": $(mobjArr1[7]).val(),
            "YQMC3": $(mobjArr1[8]).val(),
            "XH3": $(mobjArr1[9]).val(),
            "BH3" : $(mobjArr1[10]).val(),
            "ZT3" : $(mobjArr1[11]).val(),
            "YQMC4" : $(mobjArr1[12]).val(),
            "XH4": $(mobjArr1[13]).val(),
            "BH4": $(mobjArr1[14]).val(),
            "ZT4": $(mobjArr1[15]).val(),
            "YQMC5": $(mobjArr1[16]).val(),
            "XH5": $(mobjArr1[17]).val(),
            "BH5": $(mobjArr1[18]).val(),
            "ZT5": $(mobjArr1[19]).val(),
            "BZXMBHGS": $(objArr[objArr.length-2]).val(),
            "YBXMBHGS": $(objArr[objArr.length-1]).val(),
        }]
    };
    $.ajax({
        type: "post",
        url: "/AppM/addbuilding_hoist",
        data: { obj: JSON.stringify(json) },
        success: function (data) {
            layer.msg("提交成功");

        }
    })

    }
}